services/web-app/src/lib/hooks/useAxios.tsx

```diff
@@ -1,21 +1,30 @@
-import axios, { AxiosInstance } from "axios";
-import { BASE_URL } from "@/lib/constants";
+import axios, { type AxiosInstance } from "axios";
 import { getSession } from "next-auth/react";
-import { Session } from 'next-auth';
+
+import { BASE_URL } from "../constants";
 
 const axiosInstance = axios.create({
   baseURL: `${BASE_URL}`,
 });
 
-const useAxios = (): { axiosInstance: AxiosInstance } => {
-  const session = getSession();
+const useAxios = async (): Promise<{ axiosInstance: AxiosInstance }> => {
+  const session = await getSession();
+
+  if (session === null || !('token' in session)) {
+    throw new Error(
+      "Error: logged in user's session data not fetched correctly."
+    );
+  }
+
   axiosInstance.interceptors.request.clear();
-  axiosInstance.interceptors.request.use(
-    (config) => {
-      config.headers.Authorization = (session as Session).token;
-      return config;
-    }
-  );
+  axiosInstance.interceptors.request.use((config) => {
+    //TODO: investigate this as the Session type returned from does not seem to contain 'token'
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    config.headers.Authorization = session.token;
```

Comment:
[Note] I didn't have time to investigate this :(

Reply:
I'll add this to my current bug ticket :)